helm-projectile.el

```diff
@@ -368,6 +368,69 @@ CANDIDATE is the selected file.  Used when no file is explicitly marked."
                                             (list candidate))))
           (rename-buffer dired-buffer-name))))))
 
+
+(defun helm-projectile-find-file-or-marked (candidate)
```

Comment:
I would have simply advised the original function instead of copying it. :-) Add some advice that's executing the hook after it and this should be enough.